compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -346,19 +346,34 @@ impl DataFlowGraph {
         }
     }
 
+    /// Look up a value by ID.
+    fn get_value<R>(&self, value: ValueId<R>) -> &Value {
+        &self.values[value.raw()]
```

Comment:
Maybe this is missing a `resolve_value` call?

Reply:
Good question, I don't know, it's been like this before [half the time](https://github.com/noir-lang/noir/blob/3ac30be979234e2c6746cecfc16444a9dbfc5da1/compiler/noirc_evaluator/src/ssa/ir/dfg.rs#L557), the [other half](https://github.com/noir-lang/noir/blob/3ac30be979234e2c6746cecfc16444a9dbfc5da1/compiler/noirc_evaluator/src/ssa/ir/dfg.rs#L451) it resolved it. I added `resolve_value` for the latter case.

It is interesting for example that [get_value_max_num_bits](https://github.com/noir-lang/noir/blob/3ac30be979234e2c6746cecfc16444a9dbfc5da1/compiler/noirc_evaluator/src/ssa/ir/dfg.rs#L359) looks up by `ValueId` without `resolve, and considers the case when the value is a `NumericConstant`, but [get_numeric_constant_with_type](https://github.com/noir-lang/noir/blob/3ac30be979234e2c6746cecfc16444a9dbfc5da1/compiler/noirc_evaluator/src/ssa/ir/dfg.rs#L451) calls resolve 🤷